test/logic/models/content_test.dart

```diff
@@ -0,0 +1,297 @@
+      ...createVariations(null),
+      ...createVariations(true),
+      ...createVariations(0),
+      ...createVariations(double.nan),
+      ...createVariations(""),
+    ];
+    for (final key in keys) {
+      for (final wrongTypeValue in wrongTypeValues) {
+        if (this[key].runtimeType != wrongTypeValue.runtimeType) {
+          yield Map.of(this)..[key] = wrongTypeValue;
+        }
+      }
+    }
+  }
+}
+
+extension TestListMap on List<Map<String, dynamic>> {
+  List<Map<String, dynamic>> assignUniqueIds({required int startId}) {
+    for (var element in this) {
+      if (element['id'] is int) {
+        element['id'] = startId++;
+      }
+    }
+    return this;
+  }
+}
+
+void main() {
+  group('Handles invalid or incomplete values from the media store', () {
+    test('Handles invalid or incomplete albums', () async {
+      final validAlbum = albumWith().toMap();
+      final validAlbums = [
+        (validAlbum.copyWith({'id': 1}), albumWith(id: 1)),
+        (validAlbum.copyWith({'id': 2, 'album': ''}), albumWith(id: 2, album: '')),
+        (validAlbum.copyWith({'id': 3, 'albumArt': ''}), albumWith(id: 3, albumArt: '')),
+        (validAlbum.copyWith({'id': 4, 'albumArt': null}), albumWith(id: 4, albumArt: null)),
+        (validAlbum.copyWith({'id': 5}).copyWithout(['albumArt']), albumWith(id: 5, albumArt: null)),
+        (validAlbum.copyWith({'id': 6, 'artist': ''}), albumWith(id: 6, artist: '')),
+        (validAlbum.copyWith({'id': 7, 'artistId': -1}), albumWith(id: 7, artistId: -1)),
+        (validAlbum.copyWith({'id': 8, 'artistId': null}), albumWith(id: 8, artistId: null)),
+        (validAlbum.copyWith({'id': 9}).copyWithout(['artistId']), albumWith(id: 9, artistId: null)),
+        (validAlbum.copyWith({'id': 10, 'firstYear': -1}), albumWith(id: 10, firstYear: -1)),
+        (validAlbum.copyWith({'id': 11, 'firstYear': null}), albumWith(id: 11, firstYear: null)),
+        (validAlbum.copyWith({'id': 12}).copyWithout(['firstYear']), albumWith(id: 12, firstYear: null)),
+        (validAlbum.copyWith({'id': 13, 'lastYear': -1}), albumWith(id: 13, lastYear: -1)),
+        (validAlbum.copyWith({'id': 14, 'lastYear': null}), albumWith(id: 14, lastYear: null)),
+        (validAlbum.copyWith({'id': 15}).copyWithout(['lastYear']), albumWith(id: 15, lastYear: null)),
+        (validAlbum.copyWith({'id': 16, 'numberOfSongs': -1}), albumWith(id: 16, numberOfSongs: -1)),
+        (validAlbum.copyWith({'id': 17, 'numberOfSongs': null}), albumWith(id: 17, numberOfSongs: 0)),
+        (validAlbum.copyWith({'id': 18}).copyWithout(['numberOfSongs']), albumWith(id: 18, numberOfSongs: 0)),
+        (
+          validAlbum.copyWith({'id': 19, 'lastYear': 2000, 'firstYear': 3000}), // Last year before first year
+          albumWith(id: 19, lastYear: 2000, firstYear: 3000),
+        ),
+      ];
+      final propertiesThatCanBeMissing = ['albumArt', 'artistId', 'firstYear', 'lastYear', 'numberOfSongs'];
```

Comment:
If we add a new property, this will likely not be updated

Reply:
The way I see it, that's not going to be a problem. The tests verify the current behaviour, any change will (and should) break them. If we add a nullable property, the `subSets` function below will generate albums maps where that property is null/missing, so the test will fail when we expect parsing of these to fail but they get accepted.